### PR TITLE
Code Cleanup: Calculation grouping out of util and base class with testing calculation methods directly

### PR DESCRIPTION
**Pull request by Sethvg**

### CODE REVIEW

**Review by joshzarrabi — APPROVED** [review]
Looks great!

File: frontend/src/data/utils/DatasetCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      const total = totalRow.first()[rawCountCol];
+      return group
+        .generateSeries({
+          [pctShareCol]: (row) => this.percent(row[rawCountCol], total),
+        })
+        .resetIndex();
+    });
+  }
+
+  calculatePctShareOfKnown(
```
Comment (vanshkumar) [inline frontend/src/data/utils/DatasetCalculator.ts:65]: It would be good to add a test for this function as you are adding tests for all the others

**Review by vanshkumar — DISMISSED** [review]
thanks for adding testing!

File: frontend/src/data/utils/DatasetCalculator.ts (hunk 1 above)
Comment (Sethvg) [reply frontend/src/data/utils/DatasetCalculator.ts:65]: okie

**Review by vanshkumar — APPROVED** [review]

**Pull request merged**